### ISSUE TITLE
Segfault on an rtcp packet

### ISSUE THREAD

**Issue by MichaelSwann** [sipwise/rtpengine#1303]
My rtpengine is segfaulting on what looks to be on an rtcp packet according to the core dump. In our environment, this is happening multiple times a day. 

RTP Engine Version
```
Version: 9.4.1.5+0~mr9.4.1.5 git-HEAD-d5188ce
```

Kernel Version
```
Linux ct-rtc06 5.4.0-74-generic #83-Ubuntu SMP Sat May 8 02:35:39 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```

Operating System
```
lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.2 LTS
Release:	20.04
Codename:	focal
```

A systemd unit file is used to pass in the rtpengine configuration flags on startup. 

```
[Unit]
Description=rtpengine
After=syslog.target network.target local-fs.target

[Service]
; service
Type=forking
ExecStartPre=/bin/sh -c 'systemctl set-environment LOCAL_IP=123.123.123.123'
ExecStart=/usr/sbin/rtpengine --interface ${LOCAL_IP} \
--listen-ng=${LOCAL_IP}:22222 \
--listen-tcp=${LOCAL_IP}:25060 \
--listen-udp=${LOCAL_IP}:12222 \
--listen-cli=127.0.0.1:9900 \
--pidfile /run/rtpengine.pid \
--port-min 40000 \
--port-max 65000 \
--table 0 \
--silent-timeout 1800 \
--timeout 60 \
--tos 184 \
--num-threads 4 \
--log-facility local0 \
--log-facility-rtcp local1 \
--log-facility-cdr local2 \
--recording-dir /tmp \
--recording-method pcap \
--recording-format eth \
--log-level 6 \
--delete-delay 1 \
--homer 127.0.0.1:9060 \
--homer-protocol udp \
--homer-id 199
PIDFile=/run/rtpengine.pid
TimeoutSec=15s
Restart=always
; exec
User=root
Group=daemon
LimitCORE=infinity
LimitNOFILE=100000
LimitNPROC=60000
;LimitSTACK=240
LimitRTPRIO=infinity
LimitRTTIME=7000000
IOSchedulingClass=realtime
IOSchedulingPriority=2
CPUSchedulingPolicy=rr
CPUSchedulingPriority=89
UMask=0007

[Install]
```

I have attached the backtraces.

[backtrace.txt](https://github.com/sipwise/rtpengine/files/6775800/backtrace.txt)
[backtrace_full.txt](https://github.com/sipwise/rtpengine/files/6775801/backtrace_full.txt)

**Comment by rfuchs** [comment]
Please install the -dbg or -dbgsym packages and pull the backtrace again

**Comment by MichaelSwann** [comment]
I have installed the -dbgsym packets. Herewith the backtrace. 

[backtrace_full2.txt](https://github.com/sipwise/rtpengine/files/6781816/backtrace_full2.txt)

**Comment by rfuchs** [comment]
With the core loaded in gdb, can you post the output of
```
frame 6
p *pt
p * (str *) pt->rtcp_fb.head->data
p * (str *) pt->rtcp_fb.head->next->data
```

**Comment by MichaelSwann** [comment]
Herewith the result

```
(gdb) frame 6
#6  0x00005607c383d11f in insert_codec_parameters (chop=0x7fa970274b80, cm=0x7fa96808a0a0) at sdp.strhash.c:2003
2003	sdp.strhash.c: No such file or directory.

(gdb) p *pt
$1 = {payload_type = 111, encoding_with_params = {s = 0x7fa9742286a8 "opus/48000/2", len = 12}, encoding_with_full_params = {s = 0x7fa9742286b8 "opus/48000/2", len = 12}, encoding = {s = 0x7fa9742286c8 "opus", len = 4}, clock_rate = 48000,
  encoding_parameters = {s = 0x7fa9742286d0 "2", len = 1}, channels = 2, format_parameters = {s = 0x7fa9742286d8 "minptime=10;useinbandfec=1", len = 26}, codec_opts = {s = 0x0, len = 0}, rtcp_fb = {head = 0x7fa96c6cb6c0 = {0x7fa97401bdd8},
    tail = 0x7fa96c6cb6c0 = {0x7fa97401bdd8}, length = 1}, ptime = 20, bitrate = 32000, codec_def = 0x5607c38d12e0 <__codec_defs+1600>, for_transcoding = 1}

(gdb) p * (str *) pt->rtcp_fb.head->data
$2 = {s = 0x2 <error: Cannot access memory at address 0x2>, len = 1889970507}

(gdb) p * (str *) pt->rtcp_fb.head->next->data
Cannot access memory at address 0x0
```

**Comment by rfuchs** [comment]
I see the problem in the core, but I'm puzzled as to how it happens. Do you have debug logging enabled? The call signalling (offer/answer) would be interesting for this call (call ID `pkbsa4cnbhog6rgph2f2`). If not, can you enable debug logging and then post the log for the next call that triggers this?

**Comment by MichaelSwann** [comment]
I do not have debug logging enabled but can make it so.

I had to take this server out of production but will put it back into service on Monday. It crashes quite regularly so I don't expect to wait long before I can get an example.

**Comment by MichaelSwann** [comment]
Hi,

Sorry for the delay, I can only have one shot at getting an example a day. Didn't realise quite how verbose the logging is when rtpengine is in debug mode - ran out of disk space quite quickly on my first attempt. 

Herewith the backtrace and debug logs.

Backtrace
[backtrace_full-1013.txt](https://github.com/sipwise/rtpengine/files/6822721/backtrace_full-1013.txt)

The debug is quite verbose. I have grepped for the callid and will only attach those logs.
[rtpengine-kg1l6dak82v2koio6b57.log](https://github.com/sipwise/rtpengine/files/6822730/rtpengine-kg1l6dak82v2koio6b57.log)

**Comment by rfuchs** [comment]
3f00202 should be the fix for this for the m9.4 branch

**Comment by MichaelSwann** [comment]
Thank you, I will test it out.

**Comment by rfuchs** [comment]
Any feedback here?

**Comment by rfuchs** [comment]
Merged to 9.5 now, closing.

**Comment by MichaelSwann** [comment]
Apologies for the long silence from my side. I managed to contract covid which put me out of action for quite some time. 

I pushed this patch onto one of our production servers just before getting ill and it has been running ever since with no crashes.

I see the issue is closed and I want to close the loop by confirming it is working.

As always thanks for the assistance!
